packages/stark-ui/src/modules/date-range-picker/components/date-range-picker.component.ts

```diff
@@ -24,91 +38,151 @@ const componentName = "stark-date-range-picker";
 	/**
-	 * Filter function or a string
-	 * Will be applied to both date-picker
+	 * ControlValueAccessor listener
+	 * @ignore
 	 */
-	@Input()
-	public dateFilter?: StarkDatePickerFilter;
+	private _onTouched: () => void = noop;
 
 	/**
-	 * Timestamp Mask Configuration to apply on the start/end date-picker.
-	 * If `true` is passed, the default mask config is applied: {DEFAULT_DATE_MASK_CONFIG|DEFAULT_DATE_MASK_CONFIG}
-	 * If `false` is passed or if `dateMask` is not present, the directive is disabled.
-	 * If a `StarkTimestampMaskConfig` is passed, it is set as the date mask config.
+	 * ControlValueAccessor listener
+	 * @ignore
 	 */
-	@Input()
-	public dateMask?: StarkDatePickerMaskConfig;
+	private _onChange: (_dateRange: StarkDateRangePickerEvent) => void = noop;
 
 	/**
-	 * Source Date to be bound to the end datepicker model
+	 * Subscriptions to be removed at end of component lifecycle
+	 * @ignore
 	 */
-	@Input()
-	public endDate?: Date;
+	private subs: Subscription[] = [];
 
+	/*--- VIEW CHILDREN ---*/
 	/**
-	 * Label to be displayed in the end datepicker
+	 * Reference to the start datepicker embedded in this component
 	 */
-	@Input()
-	public endDateLabel = "STARK.DATE_RANGE_PICKER.TO";
+	@ViewChild("startPicker")
+	public startPicker!: StarkDatePickerComponent;
 
 	/**
-	 * Maximum date of the end date picker
+	 * Reference to the end datepicker embedded in this component
 	 */
-	@Input()
-	public endMaxDate?: Date;
+	@ViewChild("endPicker")
+	public endPicker!: StarkDatePickerComponent;
+
+	/*--- START-DATE CONFIGURATIONS ---*/
+	private _startBeforeEndVal: ValidatorFn = ({ value }) => {
+		return value instanceof Date && this.endDate instanceof Date && value.getTime() > this.endDate.getTime()
+			? { startBeforeEnd: true }
+			: null;
+	};
 
 	/**
-	 * Minimum date of the end date picker
+	 * Source Date to be bound to the start datepicker model
 	 */
 	@Input()
-	public endMinDate?: Date;
+	public get startDate(): Date | undefined {
+		return this._startDate.value || undefined;
+	}
+
+	public set startDate(value: Date | undefined) {
+		this._startDate.setValue(value);
+	}
+
+	public get startDateFormControl(): FormControl {
+		return this._startDate;
+	}
+
+	private _startDate = new FormControl(null, this._startBeforeEndVal);
 
 	/**
-	 * Whether the datepickers are disabled
+	 * Label to be displayed in the end datepicker
 	 */
 	@Input()
-	public isDisabled?: boolean;
+	public startDateLabel = "STARK.DATE_RANGE_PICKER.FROM";
 
 	/**
-	 * HTML "name" attribute of the element.
+	 * Minimum date of the start date picker
 	 */
 	@Input()
-	public rangePickerId = "";
+	public startMinDate?: Date;
 
 	/**
-	 * HTML "name" attribute of the element.
+	 * Maximum date of the start date picker
 	 */
 	@Input()
-	public rangePickerName = "";
+	public startMaxDate?: Date;
+
+	/*--- END-DATE CONFIGURATIONS ---*/
+
+	private _endAfterStartVal: ValidatorFn = ({ value }) => {
```

Comment:
Can you rename this to `_endAfterStartValidator `to make it clear it is a validator?